digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/assetUpdater/RoadLinkPropertyUpdater.scala

```diff
@@ -169,31 +225,45 @@ class RoadLinkPropertyUpdater {
+            if (overriddenAdminClass != newLink.adminClass.value && alreadyUpdatedValue.isEmpty) {
               RoadLinkOverrideDAO.insert(AdministrativeClass, newLink.linkId, Some(AutoGeneratedUsername.automaticGeneration), overriddenAdminClass)
+              transferredProperties += AdministrativeClassChange(newLink.linkId, roadLinkChangeToChangeType(change.changeType), overriddenAdminClass, Some(overriddenAdminClass))
             }
           }
         case _ => //do nothing
       }
 
       val roadLinkAttributes = LinkAttributesDao.getExistingValues(oldLink.linkId)
-      roadLinkAttributes.foreach { attribute =>
+      if (roadLinkAttributes.nonEmpty) {
         change.newLinks.foreach { newLink =>
-          LinkAttributesDao.insertAttributeValueByChanges(newLink.linkId, AutoGeneratedUsername.automaticGeneration, attribute._1, attribute._2, LinearAssetUtils.createTimeStamp())
+          val alreadyUpdatedValues = LinkAttributesDao.getExistingValues(newLink.linkId)
+          if (alreadyUpdatedValues.isEmpty) {
+            roadLinkAttributes.foreach { attribute =>
+              LinkAttributesDao.insertAttributeValueByChanges(newLink.linkId, AutoGeneratedUsername.automaticGeneration, attribute._1, attribute._2, LinearAssetUtils.createTimeStamp())
+            }
+            transferredProperties += RoadLinkAttributeChange(newLink.linkId, roadLinkChangeToChangeType(change.changeType), roadLinkAttributes, roadLinkAttributes)
+          }
         }
       }
     }
+    transferredProperties
   }
 
   def updateProperties() = {
-    val changes = roadLinkChangeClient.getRoadLinkChanges()
-    val (addChanges, remaining) = changes.partition(_.changeType == Add)
-    val (removeChanges, otherChanges) = remaining.partition(_.changeType == Remove)
-
     withDynTransaction {
-      transferOverriddenPropertiesAndPrivateRoadInfo(otherChanges)
-      transferOrGenerateFunctionalClassesAndLinkTypes(addChanges ++ otherChanges)
-      removePropertiesFromOldLinks(removeChanges ++ otherChanges)
+      val latestSuccess = Queries.getLatestSuccessfulSamuutus(RoadLinkProperties.typeId)
+      val changes = roadLinkChangeClient.getRoadLinkChanges(latestSuccess)
+      val (addChanges, remaining) = changes.partition(_.changeType == Add)
+      val (removeChanges, otherChanges) = remaining.partition(_.changeType == Remove)
+
+      val transferredProperties = transferOverriddenPropertiesAndPrivateRoadInfo(otherChanges)
+      val createdProperties = transferOrGenerateFunctionalClassesAndLinkTypes(addChanges ++ otherChanges)
+      val deletedProperties = removePropertiesFromOldLinks(removeChanges ++ otherChanges)
+      val changeReport = ChangeReport(RoadLinkProperties.typeId, transferredProperties ++ createdProperties ++ deletedProperties)
+      val (reportBody, contentRowCount) = ChangeReporter.generateCSV(changeReport)
+      ChangeReporter.saveReportToS3("roadLinkProperties", reportBody, contentRowCount)
```

Comment:
Vähän nousee mieleen tästä withDynTransaction hommasta varmaa vois joko lisätä tiedoston nimeen onnistuiko samuutus vai ei. Varmaan vois tehdä niin että tarkistaa menikö transaction läpi ja sen mukaan merkaa tiedoton alkuun tai loppuun onnistuiko samuutus vai ei. Ei nyt varmaan tässä tarvii alkaa toteuttaa. Kirjaan tonne confluese ylös idean.

Reply:
Eikös tämä toimisi vain jos samuutus failaisi yhdenkin tietolajin samuutus epäonnistumisesta? Nyt on kuitenkin ajateltu että tietolajit käsiteltäisiin erikseen. Tietysti jos ajatellaan että tää koskisi ainoastaan näitä linkin ylikirjoitettuja arvoja niin sitten voisi toimia.

Reply:
Sillee kun tietolajit samuutetaan itsenäisesti joten tuli raportti per tietolaji per päivä ja niistä jotkut teoriassa voisi epäonnistua. joten saataisiin kaikista tietolajeista joku raportti mutta joissakin olisi joku epäonnistunut tieto perässä.

Reply:
Tässä kun trancaction sisällä epäonnistuneet ja onnistuneet raportti ei erotus. Varmaan vois katsoa myöhemmin onko ongelma.